src/indicators/drought/arc2_precipitation.py

```diff
@@ -0,0 +1,884 @@
+        )
+        return directory
+
+    def _get_raw_filepath(self, main: bool) -> Path:
+        """
+        Return filepath to raw ARC2 data for specific x
+        and y bounds. All data stored within a single
+        main file. If `main`, the filepath is not
+        specified for a specific date range and designated
+        with 'main.nc' at the end. Otherwise, end of
+        filepath designated with `self.date_min` and
+        `self.date_max`.
+
+        :param main: Whether or not the filepath is
+            for the main file.
+        :param temp: Whether or not the filepath is
+            for a temp file used for saving.
+        """
+        directory = self._get_directory(RAW_DATA_DIR)
+
+        if main:
+            end = "main"
+        else:
+            end = f"{self.date_min:%d_%b_%Y}_{self.date_max:%d_%b_%Y}"
+
+        filename = (
+            f"arc2_daily_precip_{self.country_iso3}_"
+            f"{self.range_x[0]}_{self.range_x[1]}_"
+            f"{self.range_y[0]}_{self.range_y[1]}_{end}.nc"
+        )
+        return directory / filename
+
+    def _check_main_exists(self):
+        """Check if main file exists
+
+        Used to check if main file exists
+        """
+        fp = self._get_raw_filepath(main=True)
+        if not fp.exists():
+            raise OSError(
+                "Main file does not exist. "
+                "First run `download_data(main=True)`."
+            )
+
+    def load_raw_data(
+        self,
+        raw_filepath: Union[Path, None] = None,
+    ) -> pd.DataFrame:
+        """
+        Convenience function to load raw raster data, squeeze
+        it and write its CRS. The function always accesses
+        the main file if a filepath is not provided.
+
+        :param raw_filepath: Path to raw file to load. If `None`,
+            loads main file.
+        """
+        if raw_filepath is None:
+            raw_filepath = self._get_raw_filepath(main=True)
+
+        # load raw main data
+        try:
+            with rioxarray.open_rasterio(raw_filepath, masked=True) as ds:
+                da = ds.load()
+                da = da.squeeze().rio.write_crs(ARC2_CRS)
+
+        except RasterioIOError:
+            raise OSError(
+                "Raw raster file %s does not exist, "
+                "first download using `download_data()`.",
+                raw_filepath.name,
+            )
+
+        return da
+
+    def download_data(
+        self, main: bool = False, replace_missing: Union[bool, str] = False
+    ):
+        """
+        Download ARC2 data for all dates between `self.date_min`
+        and `self.date_max`. If `main`, then all data
+        is downloaded directly from the servers and set as the
+        main file. If not `main`, then data
+        is only downloaded for dates not already
+        available in the main file, and then merged
+        into the main file.
+
+        :param main: Boolean on whether to set download as
```

Comment:
Yeah, would definitely move to the clobber standard we've done elsewhere, would make it much simpler. This was when I was first just trying something out and then went haywire on it. Will shift to clobber once we drop into the main toolbox. This would just enable us to do the last option that you're talking about to keep it simple, which I really like.